__test__/ship.test.js

```diff
@@ -0,0 +1,83 @@
+const Ship = require('../src/ship')
+const Port = require('../src/port')
+const Itinerary = require('../src/itinerary')
+
+describe('ship', () => {
+
+    describe('has ports and itinerarys', () => {
+        let ship
+        let amsterdam
+        let dover
+        let itinerary 
+
+        beforeEach(() => {
+        amsterdam = {
+            addShip: jest.fn(),
+            removeShip: jest.fn(),
+            name: 'Amsterdam',
+            ships: []
+        }
+        
+        dover = {
+            addShip: jest.fn(),
+            removeShip: jest.fn(),
+            name: 'Dover',
+            ships: []
+        }
+
+        itinerary = {
+            ports: [amsterdam, dover]
+        }
+
+        ship = new Ship(itinerary)
+
+        })
+        
+        it('returns an object', () => {
+            expect(ship).toBeInstanceOf(Object)
+        })
+
+        it('it should have a starting port', () => {
+
+             expect(ship.currentPort).toBe(amsterdam)
+        })
+
+        it('can set sail', () => {
+    
+            ship.setSail()
+          
+            expect(ship.currentPort).toBeFalsy()
+            expect(amsterdam.removeShip).toHaveBeenCalledWith(ship)
+            
+          })
+          it('gets added to the port on instantiation ', () => {
+
+              expect(amsterdam.addShip).toHaveBeenCalledWith(ship)
+          })
+
+          it('can dock at a different port', () => {
+
+              ship.setSail()
+              ship.dock()
+              
+              expect(ship.currentPort).toBe(dover)
+              expect(amsterdam.removeShip).toHaveBeenCalledWith(ship)
+        })
+
+   })
+
+})
+
+describe('can\'t sail anymore', () => {
+    it('can\'t sail further than the itinerary', () => {
+        const amsterdam = new Port ('Amsterdam')
+        const dover = new Port ('Dover')
+        const itinerary = new Itinerary ([amsterdam, dover])
+        const ship = new Ship (itinerary)
+
+        ship.setSail()
+        ship.dock()
+
+        expect(() => ship.setSail()).toThrowError('Reached the end of the Itinerary!')
+    })
+})
```

Comment:
Why is this test on it's own? it doesn't need it's own describe block and the test should be moved to the other describe block. Above, you already have 2 nested describe blocks. If you move this test inside of the 1st describe block but outside of the 2nd e.g.
```
describe('ship', () => {
  describe('has ports and itinerarys', () => {
    ...
  }

  it('can\'t sail further than the itinerary', () => {
    ...
  }
})
```
you could give this test separate setup to the rest of the tests